gc/base/Configuration.cpp

```diff
@@ -437,14 +437,22 @@ MM_Configuration::initializeGCThreadCount(MM_EnvironmentBase* env)
 	MM_GCExtensionsBase* extensions = env->getExtensions();
 
 	if (!extensions->gcThreadCountForced) {
-		OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
-		extensions->gcThreadCount = omrsysinfo_get_number_CPUs_by_type(OMRPORT_CPU_TARGET);
+		extensions->gcThreadCount = supportedGCThreadCount(env);
+	}
+}
 
-		/* but not higher than maximum default */
-		if (_delegate.getMaxGCThreadCount(env) < extensions->gcThreadCount) {
-			extensions->gcThreadCount = _delegate.getMaxGCThreadCount(env);
-		}
+uintptr_t
+MM_Configuration::supportedGCThreadCount(MM_EnvironmentBase* env)
+{
+	OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
+	uintptr_t threadCount = omrsysinfo_get_number_CPUs_by_type(OMRPORT_CPU_TARGET);
+
+	/* but not higher than maximum default */
```

Comment:
this comments needs to be elaborated, and also the formatting needs to be fixed.